### ISSUE TITLE
`conda update --all` fails with `No such file or directory: '/usr/local/share/anaconda/bin/python3.6'`

### ISSUE THREAD

**Issue by ralexx** [conda/conda#5995]
Since my last `conda update` in this env, I re-ordered channel priority in `.condarc` so that conda-forge appears above defaults. No other change that I can think of.

[Full debug output](https://github.com/conda/conda/files/1322361/conda_err_20170921.txt)
[environment.yml](https://github.com/conda/conda/files/1322362/environment.yml.txt)

```
Current conda install:

               platform : osx-64
          conda version : 4.3.25
       conda is private : False
      conda-env version : 4.3.25
    conda-build version : 3.0.18
         python version : 3.6.2.final.0
       requests version : 2.14.2
       root environment : /usr/local/share/anaconda  (writable)
    default environment : /usr/local/share/anaconda
       envs directories : /usr/local/share/anaconda/envs
                          /Users/arg/.conda/envs
          package cache : /usr/local/share/anaconda/pkgs
                          /Users/arg/.conda/pkgs
           channel URLs : https://conda.anaconda.org/ralexx/osx-64
                          https://conda.anaconda.org/ralexx/noarch
                          https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://conda.anaconda.org/r/osx-64
                          https://conda.anaconda.org/r/noarch
            config file : /Users/arg/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.25 requests/2.14.2 CPython/3.6.2 Darwin/16.7.0 OSX/10.12.6
                UID:GID : 501:20

`$ /usr/local/share/anaconda/bin/conda update --all --yes --debug`




    Traceback (most recent call last):
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/core/link.py", line 326, in _execute_actions
        action.execute()
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/core/path_actions.py", line 388, in execute
        compile_pyc(python_full_path, self.source_full_path, self.target_full_path)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/gateways/disk/create.py", line 275, in compile_pyc
        subprocess_call(command, raise_on_error=False)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/gateways/subprocess.py", line 47, in subprocess_call
        p = Popen(command_arg, cwd=path, stdin=PIPE, stdout=PIPE, stderr=PIPE, env=env)
      File "/usr/local/share/anaconda/lib/python3.6/subprocess.py", line 707, in __init__
        restore_signals, start_new_session)
      File "/usr/local/share/anaconda/lib/python3.6/subprocess.py", line 1333, in _execute_child
        raise child_exception_type(errno_num, err_msg)
    FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/share/anaconda/bin/python3.6'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/core/link.py", line 281, in execute
        pkg_data, actions)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/core/link.py", line 344, in _execute_actions
        reverse_excs,
    conda.CondaMultiError: [Errno 2] No such file or directory: '/usr/local/share/anaconda/bin/python3.6'


    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/exceptions.py", line 640, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 140, in _main
        exit_code = args.func(args, p)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/cli/main_update.py", line 65, in execute
        install(args, parser, 'update')
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/cli/install.py", line 326, in install
        execute_actions(actions, index, verbose=not context.quiet)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/plan.py", line 830, in execute_actions
        execute_instructions(plan, index, verbose)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/instructions.py", line 247, in execute_instructions
        cmd(state, arg)
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/instructions.py", line 108, in UNLINKLINKTRANSACTION_CMD
        txn.execute()
      File "/usr/local/share/anaconda/lib/python3.6/site-packages/conda/core/link.py", line 297, in execute
        rollback_excs,
    conda.CondaMultiError: [Errno 2] No such file or directory: '/usr/local/share/anaconda/bin/python3.6'
```

**Comment by kalefranz** [comment]
`compile_pyc` gets hit when using noarch packages.  Are you trying to upgrade an environment from python 3.5 to python 3.6, that also includes noarch packages?  If so, then this is a known issue.

**Comment by ralexx** [comment]
No. I was performing a routine in-place `conda update --all`. Just saw the same behavior in a different env. Both envs were created with `python=3.6`.

The path to the Python 3.6 executable does exist and is executable:

```
~$ ls -l /usr/local/share/anaconda/envs/nlp36/bin/python3.6
-rwxrwxr-x  1 arg  admin  13596 Sep 15 15:50 /usr/local/share/anaconda/envs/nlp36/bin/python3.6
```

**Comment by ralexx** [comment]
I can successfully update each individual package named by `conda update --all` by specifying the package name(s) in the command `conda update foo bar baz`. This was the case in two different environments (including root).

**Comment by ralexx** [comment]
This behavior appears related to noarch packages. Is there any way to disallow `*/noarch` channels in `.condarc`? Even when I set `allow_other_channels: False` and define only `*/osx-64` channels, `conda info -a` reports that both `*/noarch` and `*/osx-64` channels are being searched.

**Comment by kalefranz** [comment]
This should already be fixed in conda 4.4 release candidates, and #6277 is also meant to address this issue.
